libraries/helper.rb

```diff
@@ -58,6 +56,59 @@ def build_user_shell
         '/usr/local/bin/bash'
       end
     end
+
+    def omnibus_toolchain_enabled?
+      # Currently we only build the Omnibus Toolchain for Solaris 10, AIX, and Nexus
+      solaris_10? || aix? || nexus?
+    end
+
+    def toolchain_full_name
+      "#{node['omnibus']['toolchain_name'] + '-' + node['omnibus']['toolchain_version']}"
+    end
+
+    def etc_shells
+      File.read('/etc/shells')
+    rescue
+      nil
+    end
+
+    def toolchain_url(toolchain_name = node['omnibus']['toolchain_name'])
+      version        = node['omnibus']['toolchain_version']
+      meta_bucket    = node['omnibus']['toolchain_meta_bucket']    # opscode-omnibus-package-metadata
+      package_bucket = node['omnibus']['toolchain_package_bucket'] # opscode-omnibus-packages
+
+      # unfortunate necessity as omni$vehicle doesn't align with ohai
+      p = {
+        os: node['os'],
+        platform_version: node['platform_version'],
+        arch: node['kernel']['machine']
+      }
+
+      case p[:arch]
+      when 'i86pc'
+        p[:arch] = 'i386'
+      when 'sun4v'
+        p[:arch] = 'sparc'
+      end
+
+      begin
+        result = JSON.parse(open("https://#{meta_bucket}.s3.amazonaws.com/#{toolchain_name}-release-manifest/#{version}.json").read)
+
+        relpath = result["#{p[:os]}"]["#{p[:platform_version]}"]["#{p[:arch]}"]["#{version}-1"]['relpath']
+        "https://#{package_bucket}.s3.amazonaws.com" + relpath
+      rescue
```

Comment:
We should specify an exact exception or this rescue could unintentionally swallow other errors.